packages/core/content-manager/server/src/controllers/collection-types.ts

```diff
@@ -390,14 +390,39 @@ export default {
         .countRelations()
         .build();
 
-      const document = id
-        ? await updateDocument(ctx, { populate })
-        : await createDocument(ctx, { populate });
+      const document = await documentManager.findOne(id, model, { populate });
 
       if (permissionChecker.cannot.publish(document)) {
         throw new errors.ForbiddenError();
       }
 
+      /**
+       * Publish can be called on two scenarios:
+       * 1. Create a new document and publish it in one request
+       * 2. Update an existing document and publish it in one request
+       *
+       * Based on user permissions:
+       * 1. User cannot create a document, but can publish
+       *    Action will be forbidden as user cannot create a document
+       * 2. User can update and publish a document
+       *    Action will be allowed, but document will not be updated, only published with the latest draft
+       */
+      const isCreate = !id;
+      const isUpdate = id;
+
+      if (isCreate && permissionChecker.cannot.create(document)) {
+        throw new errors.ForbiddenError();
+      }
+
+      if (isCreate) {
+        await createDocument(ctx);
+      }
+
+      // Update if user has update permissions
+      if (isUpdate && permissionChecker.can.update(document)) {
+        await updateDocument(ctx);
+      }
```

Comment:
What if the user doesn't have the permission? Wouldn't we want to throw?

Reply:
In this scenario we don't want to throw,

If the user doesn't have any permission , we want both Publish and Save buttons to be disabled 
![image](https://github.com/strapi/strapi/assets/20578351/2cd850c2-0bca-47df-ac73-a6c6bf99b602)

If the user does have all permissions, the "Publish" button will save and publish

But if the user only has publish permissions, we want the "Publish" button only to publish

Reply:
In the scenario where user does not have update permissions, it will not be able to update the draft either as all attribute fields will be disabled.

Reply:
Ah yes makes sense, and I see it's explained in the comment above ✨